File: 821.Shortest-Distance-to-a-Character/shortestToChar.py
"""
先确定字符在S串里的所有下标；
之后计算每个字符到下标的距离，取最小值

比较复杂
"""
class Solution:
    def shortestToChar(self, S, C):
        """
        :type S: str
        :type C: str
        :rtype: List[int]
        """
        indices = []
        for i in range(len(S)):
            if S[i] == C:
                indices.append(i)
        res = []
        for i in range(len(S)):
            dist = min([abs(i-idx) for idx in indices])
            res.append(dist)
        
        return res

File: 821.Shortest-Distance-to-a-Character/shortestToChar.cpp
/**
 * 最小值的求法：时机
 */
class Solution {
public:
    vector<int> shortestToChar(string S, char C) {
        vector<int> indices;
        vector<int> res(S.size(),S.size());
        for(int i=0;i<S.size();i++)
            if(S[i] == C) indices.push_back(i);
        
        for(int i=0;i<S.size();i++)//对每个字符，遍历计算重复下标距离的最小值
            for(int j=0;j<indices.size();j++)//
                res[i] = min(res[i], abs(indices[j]-i));//indices[i]-j计算重复下标当前下标与字符的距离，遍历求最小值
        
        return res;
    }
};
